### ISSUE TITLE
Separate active buffer highlight from vim.opt.cursorline

### ISSUE THREAD

**Issue by JoseConseco** [nvim-tree/nvim-tree.lua#2013]
**Can this functionality be implemented utilising API?**
I do not think so

**Is your feature request related to a problem? Please describe.**
If I have disabled vim.opt.cursorline then NvimTree wont highlight active buffer. 

**Describe the solution you'd like**
Custom highlight -  not dependent on nvim, but it would only depends on NvimTree config

**Describe alternatives you've considered**
We can make hack with autocmd that enables cursorline for NvimTree, but is not 'fun'.

**Comment by gegoune** [comment]
Proposal: new highlight group `NvimTreeFileCurrent` or `Focused`. Will require refreshing tree on events such as `BufEnter` or `BufWinEnter`. @alex-courtis can you see any downsides?

It does seem like very reasonable request, cursor line highlighting should be used more for cursor movement(navigation) rather than displaying state.

**Comment by alex-courtis** [comment]
This is a great idea, adds consistency. Currently there is no way to do this except for `update_focused_file`

This can be done without a refresh, a draw should be good enough as per opened files.

**Comment by gegoune** [comment]
Of course, meant redraw, not a refresh, just to update highlight group.

**Comment by alex-courtis** [comment]
This can be a `lua/nvim-tree/renderer/decorator`

**Comment by litoj** [comment]
I noticed there is currently a `NvimTreeOpenedFile` highlight group. However, I have never seen that group get used to highlight anything.

**Comment by alex-courtis** [comment]
> I noticed there is currently a `NvimTreeOpenedFile` highlight group. However, I have never seen that group get used to highlight anything.

Have you set `:help nvim-tree.renderer.highlight_opened_files` ?

**Comment by litoj** [comment]
@alex-courtis Only now that you've mentioned it, thank you. But still, the `NvimTreeOpenedFile` highlight group isn't anywhere to be seen even in the docs, so I wonder what its use is.

**Comment by alex-courtis** [comment]
> [@alex-courtis](https://github.com/alex-courtis) Only now that you've mentioned it, thank you. But still, the `NvimTreeOpenedFile` highlight group isn't anywhere to be seen even in the docs, so I wonder what its use is.

`NvimTreeOpenedFile` is the legacy highlight group from before 8.2 HIGHLIGHT: OVERHAUL, mentioned only there. See `:help nvim-tree-highlight-overhaul` and `:help nvim-tree-legacy-highlight`
The current group `NvimTreeOpenedHL` links to that legacy group only if the user has defined it.

Take a look at `:help nvim-tree-quickstart-highlight` : `:NvimTreeHiTest` to see exactly what highlighting is being used.
